### ISSUE TITLE
Don't apply messages twice when processing a chain

### ISSUE THREAD

**Issue by frrist** [filecoin-project/go-filecoin#3082]
### Description
Calling [`runMessages`](https://github.com/filecoin-project/go-filecoin/blob/0edc3e9c95f198cc5a20c3f94badb6d86160d7cc/consensus/expected.go#L349-L394) will call `ProcessBlock` for each block in a tipset and will also call `ProcessTipSet`. Both `ProcessBlock` and `ProcessTipSet` call `ApplyMessagesAndPayReward`. This means we are processing each message we receive **twice**. A solution to this is proposed in a comment block stating:
> 	// multiblock tipsets require reapplying messages to get aggregate state
	// NOTE: It is possible to optimize further by applying block validation
	// in sorted order to reuse first block transitions as the starting state
	// for the tipSetProcessor.

### Acceptance criteria
- The ProcessBlock method can be replaced by a single call to ProcessTipset, this could mean that a method similar to `ProcessBlock` is called inside `ProcessTipset`.

**Comment by ZenGround0** [comment]
Note that the comment only applies to reusing the state transition of the first block in the tipset, not doing away with `ProcessBlock` entirely.

It might be fine to process messages only once by getting rid of `ProcessBlock` as the Acceptance Criteria suggest.  I predict it's even likely to be fine, however before we get into this we absolutely need to reason about how this affects protocol correctness.  It is not obvious that something bad can't happen if we only validate the tipset.  The reason is that after the proposed change filecoin nodes would accept blocks with tipset state roots that do not match the state implied by applying all their messages to their parent state.  

The catch is that correct state roots [might not be necessary](#3083) as we don't seem to be using them for anything right now.

Even if it is ok to do this, keep in mind that we already process messages only once in the common case [where the tipset has length 1](https://github.com/filecoin-project/go-filecoin/blob/master/consensus/expected.go#L402).

**Comment by frrist** [comment]
closing as this is a now a duplicate of #2432
